Assets/Scripts/Structures/Structure.cs

```diff
@@ -165,6 +169,7 @@ public void Destroy()
             transform.DOScale(Vector3.zero, .25f).SetEase(Ease.OutSine).OnComplete(() => Destroy(gameObject));
             
             if(Manager.State.InGame) Manager.Jukebox.PlayDestroy();
+            Manager.Map.GetNeighbours(this).ForEach(neighbour => neighbour.Bonus = neighbour.AdjacencyBonus());
```

Comment:
Needs to check neighbours when building removed as it might remove their bonus

Reply:
Would it not be safer to make bonuses calculated on the fly rather than stored? More processing obviously but less error-prone.

Reply:
Had the same thinking, but because it does a lot of checking I tried to avoid that. I tested this and I think there was like 1 very specific case where it didn't work (I could only get to using debug commands), and i fixed that so I think it should be good